### PR TITLE
YML file: CI Migration to Github Actions

### LINKED ISSUE #325: Migrate CI to GitHub Actions
See context in https://github.com/scikit-learn-contrib/metric-learn/pull/322#issuecomment-919196067.

### PR DESCRIPTION
**Pull request by mvargas33**
Resolves #325 

Developed a draft YML file that handles the migration from Travis CI to Github actions.

- Python 3.6, 3.7 and 3.8 are tested with sklearn=0.20.3 + skggm for compatibility. Python 3.9 doesn't work for some reason
- Python 3.6, 3.7, 3.8 and 3.9 are tested with latest versions of dependencies, with and without skggm
- Test reports are uploaded to Codecov, as before
- Flake8 always run in the end
- All tests run on the latest Ubuntu x64

Sorry for making too many commits, I could not find a way to debug this. But only one file is added and its automatically recognized by Guthub Actions.

This proposal is already working on my fork of metric_learning, at master branch.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
```
Comment (terrytangyuan) [inline .github/workflows/main.yml:14]: This can be removed.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          pip install wheel cython numpy scipy codecov pytest-cov scikit-learn
+          pytest test --cov
+          bash <(curl -s https://codecov.io/bash)
+      - name: Run Tests with skggm
+        env:
+          SKGGM_VERSION: a0ed406586c4364ea3297a658f415e13b5cbdaf8
+        run:  |
+          pip install git+https://github.com/skggm/skggm.git@${SKGGM_VERSION}
+          pytest test --cov
+          bash <(curl -s https://codecov.io/bash)
```
Comment (terrytangyuan) [inline .github/workflows/main.yml:65]: Is this still working?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# This is a basic workflow to help you get started with Actions
```
Comment (terrytangyuan) [inline .github/workflows/main.yml:1]: Remove this

**Review by terrytangyuan — COMMENTED** [review]
Thanks! Just a couple of comments.

File: .github/workflows/main.yml (hunk 2 above)
Comment (mvargas33) [reply .github/workflows/main.yml:65]: Codecov is working, checked it.

But, idk exactly why installing skggm makes a difference for the tests. Anyway, the installation is always successful.

File: .github/workflows/main.yml (hunk 3 above)
Comment (mvargas33) [reply .github/workflows/main.yml:1]: Ok!

**Review by perimosocordiae — APPROVED** [review]
+1 to merge this after resolving @terrytangyuan's comments

File: .github/workflows/main.yml (hunk 2 above)
Comment (mvargas33) [reply .github/workflows/main.yml:64]: You use skggm for SDML, I see. As the installation is always successful, it should work. Couldn't find any error in the Github Actions logs

**Comment by mvargas33** [comment]
All comments were resolved. All tests passed on my fork. File is ready to merge
